CompuCell3D/core/CompuCell3D/plugins/MomentOfInertia/MomentOfInertiaPlugin.cpp

```diff
@@ -206,342 +186,97 @@ void CompuCell3D::MomentOfInertiaPlugin::field3DChange(const Point3D &pt, CellG
-//}
-//
-//void MomentOfInertiaPlugin::getSemiaxesXZ(CellG *_cell,double & _majorAxis , double & _medianAxis, double &_minorAxis){
-//	    //in the case of an ellipse larger moment of inertia is w.r.t. semiminor axis and is equal 1/4*a**2*M where a is semimajor axis
-//		double radical=0.5*sqrt((_cell->iXX-_cell->iZZ)*(_cell->iXX-_cell->iZZ)+4.0*_cell->iXZ*_cell->iXZ);	
-//		double lMin=0.5*(_cell->iXX+_cell->iZZ)-radical;
-//		double lMax=0.5*(_cell->iXX+_cell->iZZ)+radical;
-//		_majorAxis=2*sqrt(lMax/_cell->volume);
-//		_medianAxis=0.0;
-//	    _minorAxis=2*sqrt(lMin/_cell->volume);
-//}
-//
-//void MomentOfInertiaPlugin::getSemiaxesYZ(CellG *_cell,double & _majorAxis , double & _medianAxis, double &_minorAxis){
-//	    //in the case of an ellipse larger moment of inertia is w.r.t. semiminor axis and is equal 1/4*a**2*M where a is semimajor axis
-//		double radical=0.5*sqrt((_cell->iYY-_cell->iZZ)*(_cell->iYY-_cell->iZZ)+4.0*_cell->iYZ*_cell->iYZ);	
-//		double lMin=0.5*(_cell->iYY+_cell->iZZ)-radical;
-//		double lMax=0.5*(_cell->iYY+_cell->iZZ)+radical;
-//		_majorAxis=2*sqrt(lMax/_cell->volume);
-//		_medianAxis=0.0;
-//	    _minorAxis=2*sqrt(lMin/_cell->volume);
-//}
-//
-//
-//void MomentOfInertiaPlugin::getSemiaxes3D(CellG *_cell,double & _majorAxis , double & _medianAxis, double &_minorAxis){
-//
-//	 vector<double> aCoeff(4,0.0);	 
-//	 vector<complex<double> > roots;
-//	 //initialize coefficients of cubic eq used to find eigenvalues of inertia tensor - before pixel copy
-//	 aCoeff[0]=-1.0;
-//
-//	 aCoeff[1]=_cell->iXX + _cell->iYY + _cell->iZZ;
-//
-//	 aCoeff[2]=_cell->iXY*_cell->iXY + _cell->iXZ*_cell->iXZ + _cell->iYZ*_cell->iYZ
-//	 -_cell->iXX*_cell->iYY - _cell->iXX*_cell->iZZ - _cell->iYY*_cell->iZZ;
-//
-//	 aCoeff[3]=_cell->iXX*_cell->iYY*_cell->iZZ + 2*_cell->iXY*_cell->iXZ*_cell->iYZ
-//	 -_cell->iXX*_cell->iYZ*_cell->iYZ
-//	 -_cell->iYY*_cell->iXZ*_cell->iXZ
-//	 -_cell->iZZ*_cell->iXY*_cell->iXY;
-//
-//	 roots=solveCubicEquationRealCoeeficients(aCoeff);
-//
-//	 //finding semiaxes of the ellipsoid
-//	 //Ixx=m/5.0*(a_y^2+a_z^2) - andy cyclical permutations for other coordinate combinations
-//	 //a_x,a_y,a_z are lengths of semiaxes of the allipsoid
-//	 // We can invert above system of equations to get:
-//	vector<double> axes(3,0.0);
-//
-//	axes[0]=sqrt((2.5/_cell->volume)*(roots[1].real()+roots[2].real()-roots[0].real()));
-//	axes[1]=sqrt((2.5/_cell->volume)*(roots[0].real()+roots[2].real()-roots[1].real()));
-//	axes[2]=sqrt((2.5/_cell->volume)*(roots[0].real()+roots[1].real()-roots[2].real()));
-//
-//	//sorting semiaxes according the their lengths (shortest first)
-//	sort(axes.begin(),axes.end());
-//	_majorAxis=axes[2];
-//	_medianAxis=axes[1];
-//	_minorAxis=axes[0];
-//
-//}
-
 vector<double> MomentOfInertiaPlugin::getSemiaxes(CellG *_cell)
 {
 
 	return (this->*getSemiaxesFctPtr)(_cell);
 }
 
 vector<double> MomentOfInertiaPlugin::getSemiaxesXY(CellG *_cell){
```

Comment:
I think we are writing a bit complicated code by using iXY, iXZ etc instead of introducing inertia tensor where we could index axes by using numbers. We can think about refactoring it later. For now this is just an observation and perhaps a suggestion to add another item to our backlog